### ISSUE TITLE
Hiding threads seems to corrupt "4chan X.hiddenThreads"

### ISSUE THREAD

**Issue by Stenag** [ccd0/4chan-x#2218]
After hiding some (a lot of ?) threads the script stops loading on new pages because of an exception thrown by the JSON parser.

SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at $.getSync (userscript.html?id=0fb9b6c8-46b0-44d0-8148-790b494899e8:5346)
    at Object.init (userscript.html?id=0fb9b6c8-46b0-44d0-8148-790b494899e8:25023)
    at userscript.html?id=0fb9b6c8-46b0-44d0-8148-790b494899e8:25482
    at Window.tms_0fb9b6c8_46b0_44d0_8148_790b494899e8 (userscript.html?id=0fb9b6c8-46b0-44d0-8148-790b494899e8:25484)
    at <anonymous>:2:494
    at userscript.html?id=0fb9b6c8-46b0-44d0-8148-790b494899e8:2
    at userscript.html?id=0fb9b6c8-46b0-44d0-8148-790b494899e8:3
    at Object.window.__u__13879897.74176862 (userscript.html?id=0fb9b6c8-46b0-44d0-8148-790b494899e8:25486)
    at <anonymous>:2:494

The "hiddenThreads" line in tampermonkey storage is like this :
"4chan X.hiddenThreads": "dW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVkdW5kZWZpbmVk",

I'm using Chromium 71.0.3578.98 (64 bits), with Tampermonkey V4.7.54.

**Comment by Stenag** [comment]
More info :
Hiding threads in Paged mode seems to work all the time, while in infinite mode it may or may not corrupt the data.

**Comment by ccd0** [comment]
What version of 4chan X are you using?

This might be an issue with Tampermonkey or Chromium. Does it happen in older versions of either of those?

**Comment by Stenag** [comment]
I use 4chan X V1.14.4.7
The problem did not happen with older version, but it only started to happen yesterday like Yaanu.
I usually keep my addons up to date, and it seems that tampermonkey last update is from 2019-01-06.
Chromium next stable release is supposed to be in 2 days, so I doubt it comes from it either.

I just used version 1.14.4.3 and it had the same problem, but it took longer to trigger it. Both hiddenThreads and watchedThreads got full of dW5kZWZpbmVk.

**Comment by ccd0** [comment]
Can you think of anything that might have changed? v1.14.4.7 is from 2018-12-04 so nothing should have changed there.

**Comment by Stenag** [comment]
I think that https://github.com/ccd0/4chan-x/issues/2222 may have the same issue.
Currently, we know that it happens on Chromium, with Tampermonkey v4.7.54 and 4chan X v1.14.4.7 and 1.14.4.3, and all of them aren't the product of a recent update.
I'm not big on web development, but may it be caused by an update of some utilities called by 4chan X or 4chan ?

Currently, I've only seen Default Volume, hiddenThreads and watchedThreads getting dW5kZWZpbmVk.

**Comment by ccd0** [comment]
Are you using any other extensions? If you're using an ad blocker, can you try clearing the filter list caches and manually updating them?

**Comment by ccd0** [comment]
I've been able to reproduce this issue and #2216 on Chromium + Tampermonkey by adding
`!4chan.org##+js(set-constant.js, String.fromCharCode, noopFunc)`
to the filter list in uBlock Origin. Between https://github.com/uBlockOrigin/uAssets/commit/0bdfb5fc9bf8c61092c8f5e2b93c110cf02a5fa8 and https://github.com/uBlockOrigin/uAssets/commit/a1eeaa31dfb44061024f3681b1cb107f690e946e this filter was active on 4chan.

From my understanding this filter disrupts the decoding of a certain obfuscated anti-adblock script. The first change to the filter caused issues on 4chan including the disruption of captcha. This was fixed by removing 4chan.org from the list, since the anti-adblock script in question is no longer in use there. On other websites where this filter is still applied, there may continue to be this adverse interaction between uBlock Origin and Tampermonkey.

There's nothing I can do to recover data items converted to these strings of undefineds, but I'll see if I can revert the affected items to their default values so 4chan X starts working again for affected users.
